File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IPanelModel.java
package ru.android.zheka.coreUI;

import android.graphics.drawable.Drawable;
import android.view.View;

import androidx.databinding.ObservableField;
import androidx.databinding.ObservableInt;
import ru.android.zheka.core.IInfoModel;

public interface IPanelModel {
    int COMBO_BOX_VISIBLE = 0;
    int INPUT_TEXT_VISIBLE = 1;

    ObservableField <ButtonHandler>
            INVISIBLE_BUTTON = new ObservableField <> (new ButtonHandler ());
    ObservableField <SpinnerHandler>
            EMPTY_SPINNER = new ObservableField <> (new SpinnerHandler ());

    ObservableField <View.OnKeyListener> keyListener();

    ObservableField <String> input();

    ObservableField success();// can be translated

    ObservableField action();// can be translated

    ObservableInt progressBarVisibility();

    void progressBarVisible();

    void progressBarInvisible();

    void progressBarGone();

    ObservableInt successColor();

    ObservableField <Drawable> successImage();

    ObservableField <ButtonHandler> getStartButton();

    ObservableField <ButtonHandler> getStopButton();

    ObservableField <ButtonHandler> getNextButton();

    ObservableField <ButtonHandler> getStartButton1();

    ObservableField <ButtonHandler> getStopButton1();

    ObservableField <ButtonHandler> getNextButton1();

    ObservableField <ButtonHandler> getStartButton2();

    ObservableField <ButtonHandler> getStopButton2();

    ObservableField <ButtonHandler> getNextButton2();

    ObservableField <SpinnerHandler> getSpinner();

    ObservableInt inputVisible();

    void update(IInfoModel model);

//    void createInputKeyListener(Consumer enter, IActivity view);
}

File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IPanelModel.kt
package ru.android.zheka.coreUI

import android.graphics.drawable.Drawable
import android.view.View
import androidx.databinding.ObservableField
import androidx.databinding.ObservableInt
import ru.android.zheka.core.IInfoModel

open interface IPanelModel {
    fun keyListener(): ObservableField<View.OnKeyListener>
    fun input(): ObservableField<String>
    fun success(): ObservableField<String> // can be translated
    fun action(): ObservableField<String> // can be translated
    fun progressBarVisibility(): ObservableInt
    fun progressBarVisible()
    fun progressBarInvisible()
    fun progressBarGone()
    fun successColor(): ObservableInt
    fun successImage(): ObservableField<Drawable>
    val startButton: ObservableField<ButtonHandler>
    val stopButton: ObservableField<ButtonHandler>
    val nextButton: ObservableField<ButtonHandler>
    val startButton1: ObservableField<ButtonHandler>
    val stopButton1: ObservableField<ButtonHandler>
    val nextButton1: ObservableField<ButtonHandler>
    val startButton2: ObservableField<ButtonHandler>
    val stopButton2: ObservableField<ButtonHandler>
    val nextButton2: ObservableField<ButtonHandler>
    val spinner: ObservableField<SpinnerHandler>
    fun inputVisible(): ObservableInt
    fun update(model: IInfoModel) //    void createInputKeyListener(Consumer enter, IActivity view);

    companion object {
        const val COMBO_BOX_VISIBLE=0
        const val INPUT_TEXT_VISIBLE=1
        val EMPTY_SPINNER: ObservableField<SpinnerHandler> = ObservableField<SpinnerHandler>(SpinnerHandler())
    }
}